thunder/executors/passes.py

```diff
@@ -61,68 +62,54 @@ def preserve_bsym(bsym: BoundSymbol) -> Any:
     #   If no executor can execute the BoundSymbol, False is returned
-    def visit_helper_(bsym: BoundSymbol) -> None | bool:
-        if bsym.sym.python_impl is not None:
-            return None
-
-        ex: Executor
-        for ex in executors_list:
-            # TODO Consider allowing operator executors to claim portions of operations
-            # TODO Should FusionExecutors be allowed to claim bsym with bsym.sym.executor?
-            if (isinstance(ex, OperatorExecutor) and ex.can_execute(bsym)) or (
-                isinstance(ex, FusionExecutor) and ex.can_fuse(bsym)
-            ):
-                execution_transform: None | Callable = ex.get_execution_transform(bsym.sym)
-                out: Any
-                if execution_transform is not None:
-                    out = execution_transform(*bsym.args, **bsym.kwargs)
-                elif isinstance(ex, OperatorExecutor):
-                    # NOTE execution_transform is None and the executor is an operator executor
-                    # Calls the operator executor's operation
-                    # TODO Instead of directly acquiring the symbol from the implmap, we probably
-                    #   want to hide this behind a function
-                    op = ex.implmap[bsym.sym.id].symbol
-                    out = op(*bsym.args, **bsym.kwargs)
-                elif isinstance(ex, FusionExecutor):
-                    # NOTE execution_transform is None and the executor is a fusion executor
-                    # Preserves the symbol as is (it will be handled in the fusion pass)
-                    out = preserve_bsym(bsym)
-                else:
-                    raise AssertionError("Unknown executor")
-
-                safe_map_flat(update_swapmap, bsym.output, out)
-                return True
-
-        if bsym.sym.executor is not None:
-            return None
-
-        return False
-
-    def visit_(bsym: BoundSymbol) -> transforms.VISIT_TYPE:
-        result: None | bool = visit_helper_(bsym)
-
-        if result is None:
-            return transforms.VISIT_TYPE.NO_OP
-
-        if result is True:
-            return transforms.VISIT_TYPE.REPLACE
-
-        # NOTE result is False (which means no executor was found for the symbol)
-        cutils.check(not bsym.sym.is_prim, lambda: f"Failed to find an executor for bound symbol {bsym=}")
-        for sbsym in bsym.subsymbols:
-            visit_(sbsym)
-
-        return transforms.VISIT_TYPE.REPLACE
-
-    extrace = transforms.visitor_transform(trace, visit_)
-
-    # Restores original variables
-    bound_symbols: list[BoundSymbol] = []
-    for bsym in extrace.bound_symbols:
-        nbsym: BoundSymbol = bsym.from_bsym_swap_proxies(swapmap)
-        bound_symbols.append(nbsym)
-
-    extrace.bound_symbols = bound_symbols
+
+    class OpExProcessor(TraceSubstitutionProcessor):
+        def process_bsym(self, bsym):
+            if bsym.sym.python_impl is not None:
```

Comment:
Comment for this short-circuit would be cool

Reply:
Yeah, I copy-pasted this from your version of the visitor logic. What would be the comment?